### PR TITLE
Extract BaseQueryAssertionsTest

### PR DESCRIPTION
**Pull request by kokosing**
Extract BaseQueryAssertionsTest

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/query/BaseQueryAssertionsTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.testng.annotations.BeforeClass;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public abstract class BaseQueryAssertionsTest
+{
+    private QueryAssertions assertions;
+
+    @BeforeClass
+    public void init()
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/query/BaseQueryAssertionsTest.java]: make this `final`

File: presto-main/src/test/java/com/facebook/presto/sql/query/BaseQueryAssertionsTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    protected Session getDefaultSession()
+    {
+        return testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("default")
+                .build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/query/BaseQueryAssertionsTest.java]: make this `final`

**Comment by martint** [comment]
What’s the purpose of this change? I specifically was trying to avoid using inheritance as a way to reuse code when I wrote that.

My goal is to eventually switch it to use junit5’s runners, etc.

**Comment by kokosing** [comment]
> What’s the purpose of this change? 

To reduce code repetition. No big deal though.

> I specifically was trying to avoid using inheritance as a way to reuse code when I wrote that. My goal is to eventually switch it to use junit5’s runners, etc.

Good to know. Do you think using inheritance today, makes it more difficult to migrate to juni5's runners? To me it could be even easier because now we exactly see where junit5 runner should be used.

In some free time (if such exists) I will give a shoot to junit5 runners.

**Comment by martint** [comment]
It might not make it much more difficult, but adding inheritance encourages people to pile on more stuff into the base class over time for convenience, and that makes it harder to untangle later. By going with the current approach, it becomes clear that all those classes are unrelated to each other.

**Comment by kokosing** [comment]
> adding inheritance encourages people to pile on more stuff into the base class

True. 


I will leave this PR open for a while to remind me to play with junit 5.

**Comment by kokosing** [comment]
Superseded by #11886 11886

**Pull request closed without merging**